### PR TITLE
STP-2891 (1.0): Manage Sealed Secrets

### PR DESCRIPTION
**Pull request by nrockershousen**

### CODE REVIEW

File: operations/kubernetes/Manage_Sealed_Secrets.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
+
+
+## Manage Sealed Secrets
+
+Procedures for encrypting, decrypting, and modifying Kubernetes sealed secrets.
```
Comment (brantk-hpe) [inline operations/kubernetes/Manage_Sealed_Secrets.md:5]: It's not obvious to me that this section contains procedures for encrypting, decrypting, and modifying Kubernetes sealed secrets. I would expect there to be subsections "Encrypting sealed secrets", etc.

Note that sealed secrets isn't a Kubernetes thing. "Kubernetes" should be removed from this.

File: operations/kubernetes/Manage_Sealed_Secrets.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+Many of the examples in this section assume the USB stick used to install the system
+is still available. If site-init is no longer available on the USB stick and a backup
+has not been made, a new site-init will need to be created.
```
Comment (brantk-hpe) [inline operations/kubernetes/Manage_Sealed_Secrets.md:9]: This last sentence should make it clear where to look for how to create a new site-init.

File: operations/kubernetes/Manage_Sealed_Secrets.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+      Generating type static...
+      Generating type randstr...
+      Generating type randstr...
+      Creating Sealed Secret keycloak-users-localize
+      Generating type static...
+      ```
+
+
+
+### Prevent Regeneration of Tracked Sealed Secrets
```
Comment (brantk-hpe) [inline operations/kubernetes/Manage_Sealed_Secrets.md:128]: When does an admin do this?

File: operations/kubernetes/Manage_Sealed_Secrets.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+In order to prevent tracked sealed secrets from being regenerated, they
```
Comment (brantk-hpe) [inline operations/kubernetes/Manage_Sealed_Secrets.md:130]: This uses the term "tracked sealed secrets" before it's been defined what a tracked sealed secret is.

File: operations/kubernetes/Manage_Sealed_Secrets.md (hunk 1 above)
Comment (nrockershousen) [reply operations/kubernetes/Manage_Sealed_Secrets.md:5]: Would this section make more sense in the security_and_authentication subdirectory?

**Review by bklei — APPROVED** [review]

File: operations/kubernetes/Manage_Sealed_Secrets.md (hunk 1 above)
Comment (spillerc-hpe) [reply operations/kubernetes/Manage_Sealed_Secrets.md]: I had to modify the cray_reds_credentials secret on surtur earlier due to a typo in one of the passwords, I'll write that up into a procedure.

Comment (nrockershousen) [reply operations/kubernetes/Manage_Sealed_Secrets.md]: @spillerc-hpe do you have that information written up anywhere currently?

File: operations/security_and_authentication/Manage_Sealed_Secrets.md [hunk 5]
```diff
@@ -190,3 +191,86 @@ Expected output looks similar to the following:
...
+   {"SNMPUsername": "testuser", "SNMPAuthPassword": "testpass1", "SNMPPrivPassword": "testpass2"}
+   ```
+
+6. Replace the cray_reds_credentials secret in customizations.yaml with one containing the new credentials.
+
+   ```bash
+   ncn-m# cat cray_reds_credentials.json | ./utils/secrets-encrypt.sh | yq w -f - -i customizations.yaml 'spec.kubernetes.sealed_secrets.cray_reds_credentials'
+   ```
+
+7. Verify that customizations.yaml contains the updated password.
```
Comment (spillerc-hpe) [inline operations/security_and_authentication/Manage_Sealed_Secrets.md:273]: I don't see the final command and output for this step?
```
$ ./utils/secrets-decrypt.sh cray_reds_credentials | jq -r '.data.vault_switch_defaults' | base64 --decode
{"SNMPUsername": "testuser", "SNMPAuthPassword": "testpass1", "SNMPPrivPassword": "testpass2"}
```

Comment (nrockershousen) [reply operations/security_and_authentication/Manage_Sealed_Secrets.md:273]: Fixed.

**Review by spillerc-hpe — APPROVED** [review]

**Pull request merged**